Updated NPM modules and dependencies

hunternovak opened the pull request:
Most prevalent adjustments made for new lodash method names and aliases.

Tested against latest Hightail webapp and things seem to be working well.

easyspeak commented on lib/client/client.wilson.js:
```diff
@@ -27,6 +27,9 @@
   var _controllerProvider = null;
   var _provider           = null;
 
+  // Decorate Lodash with conveniences
```
This seems to be reoccurring. It could make sense to make a new local module 'ht-lodash' that adds these conveniences and can be included in other scripts to simplify reuse.

hunternovak replied:
yeah... For this instance on the client, its a little different since lodash is required from the application level rather than included with the client wilson code itself.  But for the server side node stuff definitely.

hunternovak replied:
Since that special classify method just re-uses lodash string methods to re-create the classify functionality, this should be safe here (just not as pretty)

easyspeak commented on lib/server/utils/UrlUtil.js:
```diff
@@ -1,4 +1,9 @@
+    ustring = require('underscore.string');
+
+
+// Decorate required functions
```
This could also be done in 'ht-lodash'

hunternovak commented:
@easyspeak I updated server code to include an HtLodash utility that is used throughout the server-side code rather than lodash.  This has the decorations on it as well as the mergeAll method that used to be on the hidash utility (which I removed.)
